homeassistant/components/xiaomi_miio/number.py

```diff
@@ -0,0 +1,184 @@
+    """Set up the Selectors from a config entry."""
+    entities = []
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        model = config_entry.data[CONF_MODEL]
+        device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+        coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+        device_features = 0
+        entity_class = None
+
+        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+        if model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
+            entity_class = XiaomiAirHumidifierNumber
+        elif model in [MODEL_AIRHUMIDIFIER_CA4]:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+            entity_class = XiaomiAirHumidifierNumber
+        elif model in MODELS_HUMIDIFIER:
+            device_features = FEATURE_FLAGS_AIRHUMIDIFIER
+            entity_class = XiaomiAirHumidifierNumber
+        else:
+            return
+
+        for feature in NUMBER_TYPES:
+            number = NUMBER_TYPES[feature]
+            if feature & device_features and feature in NUMBER_TYPES:
+                entities.append(
+                    entity_class(
+                        f"{config_entry.title} {number.name}",
+                        device,
+                        config_entry,
+                        f"{number.short_name}_{config_entry.unique_id}",
+                        number,
+                        coordinator,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class XiaomiAirHumidifierNumber(XiaomiCoordinatedMiioEntity, NumberEntity):
+    """Representation of a generic Xiaomi attribute selector."""
+
+    def __init__(self, name, device, entry, unique_id, number, coordinator):
+        """Initialize the generic Xiaomi attribute selector."""
+        super().__init__(name, device, entry, unique_id, coordinator)
+        self._attr_icon = number.icon
+        self._attr_unit_of_measurement = number.unit_of_measurement
+        self._attr_min_value = number.min
+        self._attr_max_value = number.max
+        self._attr_step = number.step
+        self._supported_features = 0
+        self._device_features = 0
+        self._state_attrs = {}
+        self._controller = number
+        self._value = self._extract_value_from_attribute(
+            self.coordinator.data, self._controller.short_name
+        )
+
+    @property
+    def value(self):
+        """Return the current numbered value."""
+        return self._value
+
+    @property
+    def available(self):
+        """Return the number controller availability."""
+        if (
+            super().available
+            and not self.coordinator.data.is_on
+            and not self._controller.available_with_device_off
+        ):
+            return False
```

Comment:
Do we really need this?

Reply:
Yes, the motor speed control is not available when the device is switched off. I have tested this.